aten/src/ATen/native/Activation.cpp

```diff
@@ -43,6 +44,218 @@ Tensor & rrelu_(Tensor & self, Scalar lower, Scalar upper, bool training, Genera
   return at::rrelu_with_noise_(self, self.type().tensor(), lower, upper, training, generator);
 }
 
+// -----------------------------------
+// prelu forward
+// -----------------------------------
+template <typename scalar_t>
+void inline prelu_cpu_kernel_multi_weights(
+  Tensor& result,
+  const Tensor& input,
+  const Tensor& weight,
+  int64_t input_stride0,
+  int64_t input_stride1) {
+
+  int64_t i;
+  int64_t input_numel = input.numel();
+  auto result_data = result.data<scalar_t>();
+  auto input_data = input.data<scalar_t>();
+  auto weight_data = weight.data<scalar_t>();
+
+  // multiply values at each channel with weight[channel]
+  #pragma omp parallel for private(i) if (input_numel > 1000)
+  for (i = 0; i < input_numel; i++) {
+    int64_t channel = (i % input_stride0) / input_stride1;
+    scalar_t input_data_val = input_data[i];
+    result_data[i] = (input_data_val > 0) ? input_data_val : weight_data[channel] * input_data_val;
+  }
+}
+
+template <typename scalar_t>
+void inline prelu_cpu_kernel_share_weights(
+  Tensor& result,
+  const Tensor& input,
+  const Tensor& weight) {
+
+  int64_t i;
+  int64_t input_numel = input.numel();
+  auto result_data = result.data<scalar_t>();
+  auto input_data = input.data<scalar_t>();
+  auto weight_val = weight.data<scalar_t>()[0];
+
+  #pragma omp parallel for private(i) if (input_numel > 1000)
+  for (i = 0; i < input_numel; i++) {
+    scalar_t input_data_val = input_data[i];
+    result_data[i] = (input_data_val > 0) ? input_data_val : weight_val * input_data_val;
```

Comment:
Be aware of https://github.com/pytorch/pytorch/pull/9206 . Looks like you are reverting an optimization here. Could you redo that and add a note?

Reply:
Very nice! Will change accordingly.